basicapp/basicapp.go

```diff
@@ -0,0 +1,252 @@
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		b.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installed chart %#q", b.chart.Name))
+	}
+
+	{
+		b.logger.LogCtx(ctx, "level", "debug", "message", "waiting for deployed status")
+
+		err = b.resource.WaitForStatus(b.chart.Name, "DEPLOYED")
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		b.logger.LogCtx(ctx, "level", "debug", "message", "chart is deployed")
+	}
+	{
+		b.logger.LogCtx(ctx, "level", "debug", "message", "checking resources")
+
+		for _, ds := range b.chartResources.DaemonSets {
+			b.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking daemonset %#q", ds.Name))
+
+			err = b.checkDaemonSet(ds)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			b.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("daemonset %#q is correct", ds.Name))
+		}
+
+		for _, d := range b.chartResources.Deployments {
+			b.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("checking deployment %#q", d.Name))
+
+			err = b.checkDeployment(ctx, d)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			b.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("deployment %#q is correct", d.Name))
+		}
+
+		b.logger.LogCtx(ctx, "level", "debug", "message", "resources are correct")
+	}
+
+	{
+		b.logger.LogCtx(ctx, "level", "debug", "message", "running release tests")
+
+		err = b.helmClient.RunReleaseTest(ctx, b.chart.Name)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		b.logger.LogCtx(ctx, "level", "debug", "message", "release tests passed")
+	}
+
+	return nil
+}
+
+// checkDaemonSet ensures that key properties of the daemonset are correct.
+func (b *BasicApp) checkDaemonSet(expectedDaemonSet DaemonSet) error {
+	ds, err := b.clients.K8sClient().Apps().DaemonSets(expectedDaemonSet.Namespace).Get(expectedDaemonSet.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return microerror.Maskf(notFoundError, "daemonset %#q", expectedDaemonSet.Name)
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = b.checkLabels("daemonset labels", expectedDaemonSet.Labels, ds.ObjectMeta.Labels)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = b.checkLabels("daemonset matchLabels", expectedDaemonSet.MatchLabels, ds.Spec.Selector.MatchLabels)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = b.checkLabels("daemonset pod labels", expectedDaemonSet.Labels, ds.Spec.Template.ObjectMeta.Labels)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// checkDeployment ensures that key properties of the deployment are correct.
+func (b *BasicApp) checkDeployment(ctx context.Context, expectedDeployment Deployment) error {
+
+	o := func() error {
+		err := b.checkDeploymentReady(ctx, expectedDeployment)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		return nil
+	}
+
+	off := backoff.NewConstant(30*time.Second, 5*time.Second)
```

Comment:
```suggestion
	bo := backoff.NewConstant(30*time.Second, 5*time.Second)
```

Convention is to use b but it collides. WDYT to this?

Reply:
bo! nice word 😆